runtime/java/src/common/CopperUtil.java

```diff
@@ -0,0 +1,240 @@
+package common;
+
+import common.javainterop.ConsCellCollection;
+import edu.umn.cs.melt.copper.compiletime.spec.grammarbeans.*;
+import edu.umn.cs.melt.copper.compiletime.spec.grammarbeans.visitors.ParserSpecProcessor;
+import edu.umn.cs.melt.copper.main.CopperIOType;
+import edu.umn.cs.melt.copper.main.CopperPipelineType;
+import edu.umn.cs.melt.copper.main.ParserCompiler;
+import edu.umn.cs.melt.copper.main.ParserCompilerParameters;
+import edu.umn.cs.melt.copper.runtime.engines.semantics.VirtualLocation;
+import edu.umn.cs.melt.copper.runtime.io.Location;
+import edu.umn.cs.melt.copper.runtime.logging.CopperException;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import silver.core.NIOVal;
+
+public final class CopperUtil {
+  private static Location LOCATION = new VirtualLocation("<silver>", -1, -1);
+
+  public static NIOVal compile(ParserBean parser, String packageName,
+                               String parserName, String outFile,
+                               Boolean dumpHtml, String dumpHtmlTo,
+                               IOToken tok) {
+    ParserCompilerParameters params = new ParserCompilerParameters();
+    params.setPackageName(packageName);
+    params.setParserName(parserName);
+    params.setOutputFile(new java.io.File(outFile));
+    params.setOutputType(CopperIOType.FILE);
+    params.setUsePipeline(CopperPipelineType.GRAMMARBEANS);
+
+    try {
+      ParserSpecProcessor.normalizeParser(parser, null);
+      return tok.wrap(ParserCompiler.compile(parser, params));
+    } catch (CopperException exc) {
+      throw new RuntimeException(exc);
```

Comment:
Should we be throwing `SilverInternalError` here instead of a generic `RuntimeException`?

Reply:
Fixed in a8cbde9a1cc5449c31961ae4f90e728335e8c31f